### DISCUSSION TITLE
Netbox fails to load table view If I set custom field with multiple object

### DISCUSSION THREAD

**Discussion by ersoyugur** [netbox-community/netbox#11398] [Q&A]
Hello,

I created a custom field with a multiple object type and ipam prefix model. Because I'd like to see this custom field values within table view under prefixes I add it under table configurations. After doing this, netbox fails to load table and throws an error. If custom field type is single object, it works fine. Am I missing something?

```
<class 'TypeError'>

'int' object is not iterable

Python version: 3.9.5
NetBox version: 3.2.5
```

**Comment by kkthxbye-code** [comment]
Can you replicate this? Did you try on the newest version? Did you edit the custom field at any point? If yes, was there already data in the custom field? What is the full error with stack trace (enable debugging)? Did you set any values for the custom field via. API or script?

**Reply by ersoyugur** [reply]
deleting and recreating custom field with multiple object type at first solved the issue. What I did was creating custom field with single object type, but later I changed it to multiple objects. I think this is where it started. It also broke prefixes.all() function within pynetbox. When I try to pull prefixes with api it threw the error below. I found similar issue on https://github.com/netbox-community/netbox/issues/4659. This page also where I found workaround.

`pynetbox.core.query.RequestError: The request failed with code 500 Internal Server Error but more specific details were not returned in json. Check the NetBox Logs or investigate this exception's error attribute.`

I'll try to replicate issue on both 3.2.5 and latest docker version and let you know.

**Reply by kkthxbye-code** [reply]
There's currently no cleanup logic for changing the type of a custom field, so if you have data from the previous type it will mess up when changing it to a different type. I recommend always deleting and recreating custom fields to avoid this issue.

Might be worth changing this so that it's not possible to change the type of an existing custom field.
